clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java

```diff
@@ -289,41 +288,191 @@ public void testAppend() throws IOException {
         fs.append(null, 0, null);
     }
 
-    /*
+    private void mockNonExistingPath(ObjectLocation objectLoc) throws ApiException {
+        when(objectsApi.statObject(objectLoc.getRepository(), objectLoc.getRef(), objectLoc.getPath()))
+                .thenThrow(new ApiException(HttpStatus.SC_NOT_FOUND, "no such file"));
+        when(objectsApi.listObjects(objectLoc.getRepository(), objectLoc.getRef(),
+                objectLoc.getPath() + Constants.SEPARATOR, "", 1, ""))
+                .thenReturn(new ObjectStatsList());
+    }
+
+    private void mockExistingDirPath(ObjectLocation dirObjLoc, ObjectLocation fileInDir) throws ApiException {
+        when(objectsApi.statObject(dirObjLoc.getRepository(), dirObjLoc.getRef(), dirObjLoc.getPath()))
+                .thenThrow(new ApiException(HttpStatus.SC_NOT_FOUND, "no such file"));
+
+        ObjectStats fileStat = mockExistingFilePath(fileInDir);
+        ObjectStatsList filesInDir = new ObjectStatsList();
+        filesInDir.addResultsItem(fileStat).setPagination(new Pagination().hasMore(false));
+
+        when(objectsApi.listObjects(dirObjLoc.getRepository(), dirObjLoc.getRef(),
+                dirObjLoc.getPath() + Constants.SEPARATOR, "", 1, ""))
+                .thenReturn(filesInDir);
+        when(objectsApi.listObjects(dirObjLoc.getRepository(), dirObjLoc.getRef(),
+                dirObjLoc.getPath() + Constants.SEPARATOR, "", 1000, ""))
+                .thenReturn(filesInDir);
+    }
+
+    private ObjectStats mockExistingFilePath(ObjectLocation objectLoc) throws ApiException {
+        String key = objectLocToS3ObjKey(objectLoc);
+        ObjectStats srcStats = new ObjectStats()
+                .path(objectLoc.getPath())
+                .sizeBytes(FILE_SIZE)
+                .mtime(MTIME)
+                .pathType(PathTypeEnum.OBJECT)
+                .physicalAddress(s3Url(key))
+                .checksum(UNUSED_CHECKSUM);
+        when(objectsApi.statObject(objectLoc.getRepository(), objectLoc.getRef(), objectLoc.getPath())).thenReturn(srcStats);
+        return srcStats;
+    }
+
+    private String objectLocToS3ObjKey(ObjectLocation objectLoc) {
+        return String.format("/%s/%s/%s",objectLoc.getRepository(), objectLoc.getRef(), objectLoc.getPath());
+    }
+
+
+    private void verifyObjDeletion(ObjectLocation srcObjLoc) throws ApiException {
+        verify(objectsApi).deleteObject(eq(srcObjLoc.getRepository()), eq(srcObjLoc.getRef()), eq(srcObjLoc.getPath()));
```

Comment:
Why do we need matchers if they are all `eq()`?
```suggestion
        verify(objectsApi).deleteObject(srcObjLoc.getRepository(), srcObjLoc.getRef(), srcObjLoc.getPath());
```

Reply:
I don't think that this function abstracts anything and actually _prefer_ the explicit version.